autograder-core/pom.xml

```diff
@@ -11,14 +11,16 @@
 
     <artifactId>autograder-core</artifactId>
     <name>autograder-core</name>
-    <description>Core checking code of the Autograder</description>
+    <description>Core checks of the autograder</description>
     <url>https://github.com/Feuermagier/autograder/autograder-core</url>
 
-    <properties>
-        <fluent.version>0.70</fluent.version>
-    </properties>
-
     <dependencies>
+        <!-- Internal -->
+        <dependency>
```

Comment:
Maybe you can set the dependency scope of the API to provided. As far as I know (but I have to check that) that would mean that the API is expected to be there and therefore it will not be directly included as transitive dependency.

-> You have to include it in autograder-cmd
-> You can omit it in autograder-full -> thus it will be excluded in the final JAR of autograder full -> only one version of the api will be loaded in intelligrade by construction (because the api is not included in the full jar)